### ISSUE TITLE
Duplicate entityKeys

### ISSUE THREAD

**Issue by jarno-steeman** [gaearon/normalizr#10]
This is my Article structure 
```javascript
{
	Article: {
		id: 1,
		text: '...',
		AuthorId: 1 // -> User
	}
}
```
If I query my database, and include the Author relation, I'd end with something like this:
```javascript
{
	Article: {
		id: 1,
		text: '...',
		AuthorId: 1,
		Author: {
			id: 1,
			name: '...'
		}
	}
}
```
All peachy, but normalizr would turn something like that into this:
```javascript
{
	Article: {
		id: 1,
		text: '...',
		AuthorId: 1,
		Author: 1
	}
}
```
As my response objects grow, these extra fields strart to add up if you have allot of relations. 

If you'd be running with a document database like mongo you'll probably not have an AuthorId defined in the first place, but this is different for lets say a MySQL database. It would be awesome if there was an option to disable this piece of code:

```javascript
if (!bag[entityKey]) {
    bag[entityKey] = {};
}
```

**Comment by gaearon** [comment]
What do you suggest the normalized output should look instead in your case?

**Comment by jarno-steeman** [comment]
Without the Author key i'd say. If you work with some kind of models library you'd have the AuthorId field defined in there and use that anyway.

**Comment by gaearon** [comment]
Are you running normalizr on server? Trying to understand the use case better..

**Comment by jarno-steeman** [comment]
Ahh yes, I am. Forgot to mention that.

**Comment by gaearon** [comment]
I'd accept a PR that lets you specify a third optional `options` argument to `normalize` call.
The only supported option would be `assignEntity`.

It default implementation is

```js
function assignEntity(obj, key, val) {
  obj[key] = val;
}
```

It would be called from [this line](https://github.com/gaearon/normalizr/blob/22c90e2e441444ae54a411c9aff137522ac8bc55/index.js#L13) instead of the direct assignment a la

```js
// before:
// normalized[prop] = visit(obj[prop], schema[prop], bag);

var entity = visit(obj[prop], schema[prop], bag);
assignEntity(normalized, prop], entity);
```

(and you'd need to drag `assignEntity` into that function from the main `normalize` call)

For your use case, you'd implement something like

```js
function assignEntity(obj, key, val) {
  obj[key] = val;
  delete obj[key + 'Id'];
}
```

If you do decide to PR this, please include tests.

Thanks!

**Comment by jarno-steeman** [comment]
Im not quite sure about how you think it should be implemented. But this change on index.js:13 made it happen for me:
```
-      normalized[prop] = visit(obj[prop], schema[prop], bag);
+      var val = visit(obj[prop], schema[prop], bag);
+      if (! schema[prop] || ! schema._virtual[prop])
+        normalized[prop] = val;
```

The option is saved on the schema instead of passed along the normalize call. That made the most sense to me when defining a schema/relations. This also enables you to specify which relation-keys you'd like to omit instead of just all of them for the entire normalize call.

Option name on the schema is called `virtual`, but if you can think of something that makes more sense...

Im passing the `virtual` option along as a second argument on `Schema.define`. That's so you can use it to call define a second time with the `virtual` option to true to only have those keys in `nestedSchema` be set virtual.

I'll send a PR (with test) if it's all good.

**Comment by gaearon** [comment]
>Bandwidth is precious!

To be fair gzip helps a lot with repetitive JSON so make sure to measure the savings with gzip enabled.

>The option is saved on the schema instead of passed along the normalize call. That made the most sense to me when defining a schema/relations. This also enables you to specify which relation-keys you'd like to omit instead of just all of them for the entire normalize call.

Does this distinction really help? It sounds to me, you'd either have one consistent format, or another consistent format..

I can't promise to accept a PR but please send one to keep the discussion going.

**Comment by jarno-steeman** [comment]
Im running with websockets though, which dont have support for that.

I had use for the option up until a few days ago when my project's database underwent some changes which made it kinda obsolete now. I'll change it to be passed along the normalize call.

**Comment by gaearon** [comment]
This is available in 1.0 via #14.
